Body is not working in response

I have Result payload with response_string attribute.
![image](https://user-images.githubusercontent.com/7725174/41767306-e3dfeba6-7626-11e8-8c1d-ad1e9b6a4840.png)

`Response(StatusOK, func() {
				Header("content_type:Content-Type")
				Body("response_string")
			})`

`tracker/gen/http/form/server/encode_decode.go:199:14: res.ResponseString undefined (type *views.FormPage has no field or method ResponseString)
`

--- a/http/codegen/client_body_types_test.go
+++ b/http/codegen/client_body_types_test.go
@@ -44,6 +44,8 @@ func TestBodyTypeInit(t *testing.T) {
 		{"body-path-user-validate", testdata.PayloadBodyPathUserValidateDSL, 2, BodyPathUserValidateInitCode},
 		{"body-primitive-array-user-validate", testdata.PayloadBodyPrimitiveArrayUserValidateDSL, 2, BodyPrimitiveArrayUserValidateInitCode},
 		{"result-body-user", testdata.ResultBodyObjectHeaderDSL, 2, ResultBodyObjectHeaderInitCode},
+		{"result-explicit-body-primitive", testdata.ExplicitBodyPrimitiveResultMultipleViewsDSL, 1, ExplicitBodyPrimitiveResultMultipleViewsInitCode},
+		{"result-explicit-body-user-type", testdata.ExplicitBodyUserResultMultipleViewsDSL, 2, ExplicitBodyUserResultMultipleViewsInitCode},
 	}
 	for _, c := range cases {
 		t.Run(c.Name, func(t *testing.T) {
@@ -119,5 +121,36 @@ func NewMethodBodyObjectHeaderMethodBodyObjectHeaderResultOK(body *MethodBodyObj
 	}
 	v.B = b
 	return v
+}
+`
+
+const ExplicitBodyPrimitiveResultMultipleViewsInitCode = `// NewMethodExplicitBodyPrimitiveResultMultipleViewResulttypemultipleviewsOK
+// builds a "ServiceExplicitBodyPrimitiveResultMultipleView" service
+// "MethodExplicitBodyPrimitiveResultMultipleView" endpoint result from a HTTP
+// "OK" response.
+func NewMethodExplicitBodyPrimitiveResultMultipleViewResulttypemultipleviewsOK(body string, c *string) *serviceexplicitbodyprimitiveresultmultipleviewviews.ResulttypemultipleviewsView {
+	v := body
+	res := &serviceexplicitbodyprimitiveresultmultipleviewviews.ResulttypemultipleviewsView{
+		A: &v,
+	}
+	res.C = c
+	return res
+}
+`
+
+const ExplicitBodyUserResultMultipleViewsInitCode = `// NewMethodExplicitBodyUserResultMultipleViewResulttypemultipleviewsOK builds
+// a "ServiceExplicitBodyUserResultMultipleView" service
+// "MethodExplicitBodyUserResultMultipleView" endpoint result from a HTTP "OK"
+// response.
+func NewMethodExplicitBodyUserResultMultipleViewResulttypemultipleviewsOK(body *UserType, c *string) *serviceexplicitbodyuserresultmultipleviewviews.ResulttypemultipleviewsView {
+	v := &serviceexplicitbodyuserresultmultipleviewviews.UserTypeView{
+		X: body.X,
+		Y: body.Y,
+	}
+	res := &serviceexplicitbodyuserresultmultipleviewviews.ResulttypemultipleviewsView{
+		A: v,
+	}
+	res.C = c
+	return res
 }
 `

--- a/http/codegen/client_types.go
+++ b/http/codegen/client_types.go
@@ -207,7 +207,7 @@ func {{ .Name }}({{- range .ClientArgs }}{{ .Name }} {{ .TypeRef }}, {{ end }}) 
 		{{- if .ReturnIsStruct }}
 			{{- range .ClientArgs }}
 				{{- if .FieldName }}
-			v.{{ .FieldName }} = {{ if .Pointer }}&{{ end }}{{ .Name }}
+			{{ if $.ReturnTypeAttribute }}res{{ else }}v{{ end }}.{{ .FieldName }} = {{ if .Pointer }}&{{ end }}{{ .Name }}
 				{{- end }}
 			{{- end }}
 		{{- end }}

--- a/http/codegen/server.go
+++ b/http/codegen/server.go
@@ -1026,7 +1026,7 @@ const responseT = `{{ define "response" -}}
 		{{- if .ServerBody.Init }}
 	body := {{ .ServerBody.Init.Name }}({{ range .ServerBody.Init.ServerArgs }}{{ .Ref }}, {{ end }})
 		{{- else }}
-	body := res{{ if .ResultAttr }}.{{ .ResultAttr }}{{ end }}
+	body := res{{ if $.ViewedResult }}.Projected{{ end }}{{ if .ResultAttr }}.{{ .ResultAttr }}{{ end }}
 		{{- end }}
 	{{- end }}
 	{{- range .Headers }}

--- a/http/codegen/service_data.go
+++ b/http/codegen/service_data.go
@@ -1217,6 +1217,7 @@ func buildResponseResultInit(resp *httpdesign.HTTPResponseExpr, e *httpdesign.En
 		code       string
 		origin     string
 		err        error
+		pointer    bool
 		clientArgs []*InitArgData
 
 		svc = sd.Service
@@ -1234,11 +1235,13 @@ func buildResponseResultInit(resp *httpdesign.HTTPResponseExpr, e *httpdesign.En
 		// attribute in the result type for body transformation.
 		if o, ok := resp.Body.Metadata["origin:attribute"]; ok {
 			origin = o[0]
+			pointer = respBody.IsPrimitivePointer(origin, true)
 			respBody = design.AsObject(respBody.Type).Attribute(origin)
 		}
 		ref := "body"
 		if design.IsObject(resp.Body.Type) {
 			ref = "&body"
+			pointer = false
 		}
 		var vcode string
 		if ut, ok := resp.Body.Type.(design.UserType); ok {
@@ -1295,14 +1298,15 @@ func buildResponseResultInit(resp *httpdesign.HTTPResponseExpr, e *httpdesign.En
 	status := codegen.Goify(http.StatusText(resp.StatusCode), true)
 	name := fmt.Sprintf("New%s%s%s", codegen.Goify(md.Name, true), codegen.Goify(md.Result, true), status)
 	return &InitData{
-		Name:                name,
-		Description:         fmt.Sprintf("%s builds a %q service %q endpoint result from a HTTP %q response.", name, svc.Name, e.Name(), status),
-		ClientArgs:          clientArgs,
-		ReturnTypeName:      svc.Scope.GoFullTypeName(result, pkg),
-		ReturnTypeRef:       svc.Scope.GoFullTypeRef(result, pkg),
-		ReturnIsStruct:      design.IsObject(result.Type),
-		ReturnTypeAttribute: codegen.Goify(origin, true),
-		ClientCode:          code,
+		Name:                     name,
+		Description:              fmt.Sprintf("%s builds a %q service %q endpoint result from a HTTP %q response.", name, svc.Name, e.Name(), status),
+		ClientArgs:               clientArgs,
+		ReturnTypeName:           svc.Scope.GoFullTypeName(result, pkg),
+		ReturnTypeRef:            svc.Scope.GoFullTypeRef(result, pkg),
+		ReturnIsStruct:           design.IsObject(result.Type),
+		ReturnTypeAttribute:      codegen.Goify(origin, true),
+		ReturnIsPrimitivePointer: pointer,
+		ClientCode:               code,
 	}
 }
 

--- a/http/codegen/testdata/result_encode_functions.go
+++ b/http/codegen/testdata/result_encode_functions.go
@@ -721,7 +721,7 @@ func EncodeMethodExplicitBodyPrimitiveResultMultipleViewResponse(encoder func(co
 		res := v.(*serviceexplicitbodyprimitiveresultmultipleviewviews.Resulttypemultipleviews)
 		w.Header().Set("goa-view", res.View)
 		enc := encoder(ctx, w)
-		body := res.A
+		body := res.Projected.A
 		if res.Projected.C != nil {
 			w.Header().Set("Location", *res.Projected.C)
 		}

